storage/series.go

```diff
@@ -283,39 +283,46 @@ func (s *dbSeries) drainStream(blocks m3db.DatabaseSeriesBlocks, stream io.Reade
 	return nil
 }
 
-func (s *dbSeries) Bootstrap(rs m3db.DatabaseSeriesBlocks, cutover time.Time) error {
-	if success, err := tryBootstrap(&s.RWMutex, &s.bs, "series"); !success {
-		return err
-	}
+// NB(xichen): we are holding a big lock here to drain the in-memory buffer.
+// This could potentially be expensive in that we might accumulate a lot of
+// data in memory during bootstrapping. If that becomes a problem, we could
+// bootstrap in batches, e.g., drain and reset the buffer, drain the streams,
+// then repeat, until len(s.pendingBootstrap) is below a given threshold.
+func (s *dbSeries) Bootstrap(rs m3db.DatabaseSeriesBlocks, cutOver time.Time) error {
```

Comment:
Why the change from `cutOver` to `cutover` out of interest?  Cutover seems to be a single word: 
http://www.dictionary.com/browse/cutover